godtools/Views/TractElements/TractInput.swift

```diff
@@ -7,15 +7,141 @@
 //
 
 import UIKit
+import SWXMLHash
 
 class TractInput: BaseTractElement {
-
-    /*
-    // Only override draw() if you perform custom drawing.
-    // An empty implementation adversely affects performance during animation.
-    override func draw(_ rect: CGRect) {
-        // Drawing code
+    
+    // MARK: - Positions and Sizes
+    
+    var xMargin: CGFloat {
+        return TractCard.xPaddingConstant
+    }
+    
+    var yMargin : CGFloat {
+        return self.properties.yMargin
+    }
+    
+    var xPosition: CGFloat {
+        return self.xMargin
+    }
+    
+    var yPosition: CGFloat {
+        return self.yStartPosition + self.yMargin
+    }
+    
+    var textViewWidth: CGFloat {
+        return self.properties.width > self.width ? self.width : self.properties.width
+    }
+    
+    var textViewHeight: CGFloat {
+        return self.properties.height
+    }
+    
+    override var width: CGFloat {
+        return super.width - self.xPosition - self.xMargin
+    }
+    
+    override var height: CGFloat {
+        get {
+            return super.height + self.yMargin + self.textViewHeight
+        }
+        set {
+            super.height = newValue
+        }
+    }
+    
+    var textViewXPosition: CGFloat {
+        return (self.width - self.textViewWidth) / 2
+    }
+    
+    var textViewYPosition: CGFloat {
+        return super.height + self.yMargin
+    }
+    
+    override func yEndPosition() -> CGFloat {
+        return self.yPosition + self.height + self.yMargin
+    }
+    
+    override func textYPadding() -> CGFloat {
+        return (self.parent?.textYPadding())!
+    }
+    
+    // MARK: - Object properties
+    
+    var properties = TractTextFieldProperties()
+    var textField = GTTextField()
+    
+    // MARK: - Setup
+    
+    override func setupElement(data: XMLIndexer, startOnY yPosition: CGFloat) {
+        self.yStartPosition = yPosition
+        let dataContent = splitData(data: data)
+        
+        var elements = [XMLIndexer]()
+        for node in data.children {
+            let nodeContent = splitData(data: node)
+            
+            if nodeContent.kind == "label" {
+                elements.append(node)
+            } else if nodeContent.kind == "placeholder" {
+                self.properties.placeholder = node["content:text"].element?.text
+            }
+        }
+        
+        buildChildrenForData(elements)
+        setupView(properties: dataContent.properties)
+    }
+    
+    override func setupView(properties: [String: Any]) {
+        loadElementProperties(properties: properties)
+        
+        self.textField.cornerRadius = self.properties.cornerRadius
+        self.textField.borderColor = self.properties.color
+        self.textField.borderWidth = self.properties.borderWidth
+        self.textField.backgroundColor = self.properties.backgroundColor
+        self.textField.placeholderTranslationKey = self.properties.placeholder!
```

Comment:
No! thanks! Just changed it to be: `self.textField.placeholderTranslationKey = self.properties.placeholder ?? ""` it is working. Are you agree with this change? The commit: https://github.com/CruGlobal/godtools-swift/pull/18/commits/666b5eca8319221e6f6b5cd6a2be77f5df6f5306